buildpack/build_test.go

```diff
@@ -812,9 +812,92 @@ func testBuild(kind string) func(t *testing.T, when spec.G, it spec.S) {
 									})
 								})
 
+								when("when bp api >= .10", func() {
+									it.Before(func() {
+										h.SkipIf(t, kind == buildpack.KindExtension, "")
+										descriptor.API = "0.10"
+									})
+
+									it("does not allow string command", func() {
+										h.Mkfile(t,
+											"[[processes]]\n"+
+												`command = "some-cmd"`,
+											filepath.Join(appDir, "launch-A-v1.toml"),
+										)
+										_, err := descriptor.Build(buildpack.Plan{}, config, mockEnv)
+										h.AssertError(t, err, "toml: incompatible types: TOML key \"processes.command\" has type string; destination has type slice")
+									})
+
+									it("writes extra commands as args before defined args", func() {
+										h.Mkfile(t,
+											"[[processes]]\n"+
+												`command = ["some-cmd", "cmd-arg"]`+"\n"+
+												`args = ["first-arg"]`,
+											filepath.Join(appDir, "launch-A-v1.toml"),
+										)
+										br, err := descriptor.Build(buildpack.Plan{}, config, mockEnv)
+										h.AssertNil(t, err)
+										h.AssertEq(t, len(br.Processes), 1)
+										h.AssertEq(t, br.Processes[0].Command, "some-cmd")
+										h.AssertEq(t, br.Processes[0].Args[0], "cmd-arg")
+										h.AssertEq(t, br.Processes[0].Args[1], "first-arg")
+									})
```

Comment:
In a world where platform supports overridable args - won't that same world support command as an array of strings? My assumption reading the spec and notes was that this was only required when bp > .10 and platform < vNext that supports both.

Reply:
Fixed in da6cf0efe6d3bec0b380fb7dcc1f573a69d66e45 - added `0.10` as valid buildpack API and fixed a few tests that used the old format that failed with the change.

Reply:
I started a discussion in Slack here: https://cloud-native.slack.com/archives/C033DV9EBDF/p1660064047936349

But I think my impression that the feature would be invisible to end-users was wrong. I think what's being asked for is that command-line arguments would be provided by the launcher to the process __as dictated by the buildpack api__, even though the launcher behavior is defined in the platform spec. The RFC mentions this [here](https://github.com/buildpacks/rfcs/blob/main/text/0093-remove-shell-processes.md#overriding-args-for-shell-processes). It's worth noting that if the metadata.toml schema is changing, that is a platform api change - albeit a smaller one than wholesale shell-removal from the launcher.

There is enough complexity here that it would probably benefit from some real-time discussion. I put it as a future topic for the Working Group.

Reply:
After some discussion it was decided that a separate spec PR & lifecycle PR will be created for the platform changes.